src/transformers/models/auto/configuration_auto.py

```diff
@@ -56,6 +56,7 @@
         ("deit", "DeiTConfig"),
         ("detr", "DetrConfig"),
         ("distilbert", "DistilBertConfig"),
+        ("donut", "DonutSwinConfig"),
```

Comment:
Not sure we should have this here, as it does not instantiate a Donut model if I understand correctly. For clip for instance, the `"clip"` model types goes with the whole `ClipModel`, not the text/vision part. Maybe have a `donut-swin` model typle like we did for `Data2Vec`?

Reply:
Yes, although I saw that for TrOCR, we use "trocr", even though `modeling_trocr`.py and `configuration_trocr`.py only define the model and configuration of the decoder (= text part).

Reply:
When I try to do this, `make fixup` and `make fix-copies` complain with the following error:
```
(env) nielsrogge@Nielss-MacBook-Pro transformers % make fix-copies
python utils/check_copies.py --fix_and_overwrite
Traceback (most recent call last):
  File "/Users/nielsrogge/Documents/python_projecten/transformers/src/transformers/utils/import_utils.py", line 1023, in _get_module
    return importlib.import_module("." + module_name, self.__name__)
  File "/opt/homebrew/Cellar/python@3.9/3.9.12/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/Users/nielsrogge/Documents/python_projecten/transformers/src/transformers/models/auto/configuration_auto.py", line 622, in <module>
    class AutoConfig:
  File "/Users/nielsrogge/Documents/python_projecten/transformers/src/transformers/models/auto/configuration_auto.py", line 647, in AutoConfig
    def from_pretrained(cls, pretrained_model_name_or_path, **kwargs):
  File "/Users/nielsrogge/Documents/python_projecten/transformers/src/transformers/models/auto/configuration_auto.py", line 609, in docstring_decorator
    lines[i] = _list_model_options(indent, config_to_class=config_to_class, use_model_types=use_model_types)
  File "/Users/nielsrogge/Documents/python_projecten/transformers/src/transformers/models/auto/configuration_auto.py", line 577, in _list_model_options
    lines = [
  File "/Users/nielsrogge/Documents/python_projecten/transformers/src/transformers/models/auto/configuration_auto.py", line 578, in <listcomp>
    f"{indent}- **{model_type}** -- {model_type_to_name[model_type]} ({MODEL_NAMES_MAPPING[model_type]} model)"
KeyError: 'donut-swin'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/nielsrogge/Documents/python_projecten/transformers/utils/check_copies.py", line 549, in <module>
    check_readme(args.fix_and_overwrite)
  File "/Users/nielsrogge/Documents/python_projecten/transformers/utils/check_copies.py", line 512, in check_readme
    model_names_mapping = transformers_module.models.auto.configuration_auto.MODEL_NAMES_MAPPING
  File "/Users/nielsrogge/Documents/python_projecten/transformers/src/transformers/utils/import_utils.py", line 1011, in __getattr__
    value = self._get_module(name)
  File "/Users/nielsrogge/Documents/python_projecten/transformers/src/transformers/utils/import_utils.py", line 1025, in _get_module
    raise RuntimeError(
RuntimeError: Failed to import transformers.models.auto.configuration_auto because of the following error (look up to see its traceback):
'donut-swin'
```

Reply:
Updated to `donut-swin`, similar to `Data2Vec`